app/src/main/java/org/mozilla/focus/home/HomeFragment.java

```diff
@@ -1233,6 +1323,7 @@ private void showContentPortal() {
         if (contentPanel != null) {
             contentPanel.show(true);
             if (AppConfigWrapper.hasEcommerceCoupons()) {
+                initCouponBanner(getContext());
```

Comment:
Not a blocker. Just think it would be better to move coupon banner related code into `ContentPortalView`

Reply:
Had a discussion slightly related to the moving this morning: ContentPortalView is currently treated like a dumb view and has no data loading logic in it so that's why I'm keeping it that way and leaving the loading to activity/fragment. However I should have abstract a class to keep these logics from bloating the HomeFragment. I'll se what I can do later.